src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -377,6 +478,54 @@ public void OnEventCommand(EventCommandEventArgs command)
                 }
             }
 
+            private void IncrementRefCount(string uniqueIdentifier, EventCommandEventArgs command)
+            {
+                // Could be unsafe if UniqueIdentifier protocol isn't followed
+                if (command.Arguments!.TryGetValue("UniqueIdentifier", out string? uniqueIdentifierArg))
+                {
+                    uniqueIdentifier = uniqueIdentifierArg!;
+                }
+
+                if (!_sharedSessionIds.Contains(uniqueIdentifier))
+                {
+                    _sharedSessionIds.Add(uniqueIdentifier);
+                    _sharedSessionRefCount += 1;
+                }
+            }
+
+            private bool ParseArgs<T>(IDictionary<string, string>? arguments, Func<string, T> conversion, string argumentName, T currentValue, out T parsedValue) where T : IEquatable<T>
+            {
+                if (arguments!.TryGetValue(argumentName, out string? argumentString))
+                {
+                    parsedValue = conversion(argumentString);
+                    return EqualityComparer<T>.Default.Equals(conversion(argumentString), currentValue);
+                }
+
+                Parent.Message($"Invalid {argumentName} provided. Using existing shared value {currentValue}"); // do we want to make this assumption - basically, if there's any ambiguity or an invalid parameter is provided, we set the value to be the SHARED one (instead of failing). More flexible, but do we want that?
```

Comment:
Comment inline - basically, how we historically just use the defaults for missing/bad values, we now use the SHARED values here if there's an existing session (instead of failing). It's more flexible, but I wasn't sure if that was the behavior we wanted.

Reply:
In hindsight I don't think I should have treated parse failures as default values, but I'm not trying to change the precedent now. I don't think it causes any significant problem and well-behaved clients shouldn't send data that doesn't parse.

I think it is fine if the session explicitly sets sessionId=SHARED and then doesn't explicitly set other parameters (or provides values that don't parse) then you use the existing value for the shared session. If the new session does explicitly set any value to something that doesn't match the existing shared session then that should report the error about incompatible config.

Reply:
Nit: If the argument is missing I would have the message say:
```suggestion
                Parent.Message($"No {argumentName} provided. Using existing shared value {currentValue}");
```